src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectDependencyProvider.cs

```diff
@@ -0,0 +1,155 @@
+        }
+
+        public IEnumerable<DependencyDescription> GetPackageDependencies()
+        {
+            var packageItems = _projectInstance.GetItems(PackageDependencyItemType);
+            var packageInfo = new Dictionary<string, DependencyDescription>();
+            if (packageItems != null)
+            {
+                foreach (var packageItem in packageItems)
+                {
+                    var packageDependency = CreateDependencyDescriptionFromItem(packageItem);
+                    if (packageDependency != null)
+                    {
+                        packageInfo[packageItem.EvaluatedInclude] = packageDependency;
+                    }
+                }
+
+                // 2nd pass to populate dependencies;
+
+                PopulateDependencies(packageInfo, packageItems);
+            }
+
+            return packageInfo.Values;
+        }
+
+
+        public IEnumerable<ResolvedReference> GetResolvedReferences()
+        {
+            var refItems = _projectInstance.GetItems(ResolvedReferenceItemType);
+            var resolvedReferences = new List<ResolvedReference>();
+            if (refItems != null)
+            {
+                foreach (var refItem in refItems)
+                {
+                    resolvedReferences.Add(CreateResolvedReferenceFromProjectItem(refItem));
+                }
+            }
+
+            return resolvedReferences;
+        }
+
+        private static ResolvedReference CreateResolvedReferenceFromProjectItem(ProjectItemInstance item)
+        {
+            string itemSpec = string.Empty;
+            string nugetPackageSource = string.Empty;
+            string version = string.Empty;
+            string resolvedPath = string.Empty;
+            string name = string.Empty;
+
+            resolvedPath = item.EvaluatedInclude;
+
+            name = resolvedPath ?? Path.GetFileNameWithoutExtension(resolvedPath);
+
+            foreach (var m in item.Metadata)
+            {
+                if (ItemSpecKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    itemSpec = m.EvaluatedValue;
+                    continue;
+                }
+
+                if (NugetSourceTypeKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    nugetPackageSource = m.EvaluatedValue;
+                    continue;
+                }
+
+                if (VersionKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    version = m.EvaluatedValue;
+                }
+            }
+
+            var isFromNugetPackage = "Package".Equals(nugetPackageSource, StringComparison.OrdinalIgnoreCase);
+            return new ResolvedReference(name, version, resolvedPath, isFromNugetPackage);
+        }
+
+        private static DependencyDescription CreateDependencyDescriptionFromItem(ProjectItemInstance item)
+        {
+            var version = item.GetMetadataValue("Version");
```

Comment:
Do all this resolution after your `if (string.IsNullOrEmpty(name)) return null` check